### PR TITLE
Sort patches and wavetables alphabetically

### PR DESCRIPTION
**Pull request by jsakkine**
Sort patches and wavetables alphabetically as described in #183.

### CODE REVIEW

File: src/common/SurgeStorage.cpp [hunk 1]
```diff
@@ -279,6 +280,36 @@ void SurgeStorage::refresh_patchlist()
...
+                         patch_category[i2].name.c_str()) < 0;
+      };
+
+   int groups[4] = {0, firstThirdPartyCategory, firstUserCategory,
+                    (int)patch_category.size()};
+
+   for (int i = 0; i < 3; i++)
+      std::sort(std::next(patchCategoryOrdering.begin(), groups[i]),
+                std::next(patchCategoryOrdering.begin(), groups[i + 1]),
+                categoryCompare);
```
Comment (jsakkine) [inline src/common/SurgeStorage.cpp]: I supposed you could also `patchCategoryOrdering.begin() + groups[i]` but for me that just looks unholy and evil.

**Comment by esaruoho** [comment]
fullscreen_06_01_2019__17_07
looks good

**Comment by jsakkine** [comment]
@esaruoho,  thanks alot!

**Comment by baconpaul** [comment]
OK reviewed the code - it looks great, thank you. Also built and tested on mac with no problem. I see nothing platform specific but getting my windows environment in order is a goal for the day also, so I'll merge this and then if windows does show a problem fix.

Thanks @jsakkine

**Pull request merged**